### ISSUE TITLE
OMG! my reproduced result of this model with poor experimental conditions is better than original results. I do not know why!

### ISSUE THREAD

**Issue by Fiona-senior** [HenryHZY/VL-PET#1]
I reproduce the result of VL-PET. Actually, I only reproduce the result of video-text: yc2c dataset. Because my experiment conditions are poor. I only have 2080Ti GPU with 12G. 

So I download the yc2c clip-vit dataset and annotations. set all the experimental environment. I turn the batchsize to 2, because of poor memory. and turn the numworker to 1. I did not change the learning rate.

So the only differences of mine and original experiments are dataset, batchsize, numworker

however, mine achieved the CIDEr score of 160.3! the original experiment CIDEr score is 153.

Is that possible? something wrong with my codes?

maybe, I think it is the problem of dataset. maybe.

I only use the dataset of yc2c to train the model. so the trained model is more suitable to yc2c.

Or is that possible that the result from poor experimental conditions can be better than  results from advanced experimental conditions?

**Comment by Fiona-senior** [comment]
I add the another dataset how2 to the yc2c, I found that is true that more dataset lowered the experiment results.  how2+yc2c I got the CIDEr result of 151 to the yc2c. However, the model becomes more generalized.

**Comment by HenryHZY** [comment]
Hi @Fiona-senior, sorry for my later reply since we are attending the conference.

As mentioned in our paper, there exist submit limitation of VALUE leaderboard. Therefore, we can only conduct video-text experiments with one seed (under multi-task setting).

For your questions:
> 1. "Actually, I only reproduce the result of video-text: yc2c dataset." & "however, mine achieved the CIDEr score of 160.3! the original experiment CIDEr score is 153."

 Do you mean the **testing result** reported from VALUE leaderboard under **single-task setting**? 
Just like your observation "I only use the dataset of yc2c to train the model. so the trained model is more suitable to yc2c.", I think this is due to overfitting of the dataset. 
And your analysis "I think it is the problem of dataset" also makes sense to me.

> 2. "Or is that possible that the result from poor experimental conditions can be better than results from advanced experimental conditions?"

Actually I don't think so. Using the same hyperparameters in different machines, the performance should be close. 
Although we may have to reduce the batchsize due to machine limitations, we can achieve similar performance with gradient accumulation.
By the way, we believe that the robustness of the experiment is very important. Facilitating other researcher to have a reference performance interval after repeating the experiment is one of the reasons why we conduct image-text experiments with multiple seeds.

> 3. "I add the another dataset how2 to the yc2c, I found that is true that more dataset lowered the experiment results. how2+yc2c I got the CIDEr result of 151 to the yc2c. However, the model becomes more generalized."

We have not explored in depth for single-task learning for video-text experiments. As you can see from paper Table8, multi-task learning does help with generalizability and learning cross-task knowledge.
But how to prevent such knowledge forgetting during the multi-task learning is really an interesting topic!

Thanks for your issue!

**Comment by Fiona-senior** [comment]
Dear Zi-Yuan Hu:

I am so so sorry about my late response. I only had the chance to check my email just now [embarrassing]. I am surprised of your response! I appreciate it very much.

Regarding your question, "Do you mean the testing result reported from VALUE leaderboard under single-task setting?"

My response is "Yes."

Actually, my previous focus was in the field of video captioning. Recently, I have developed a keen interest in the multi-task domain.

Your response is incredibly valuable to me. Thank you!

Best regards,

Sipei Gu



At 2023-10-06 14:46:20, "Zi-Yuan Hu" ***@***.***> wrote:

Hi @Fiona-senior, sorry for my later reply since we are attending the conference.

As mentioned in our paper, there exist submit limitation of VALUE leaderboard. Therefore, we can only conduct video-text experiments with one seed (under multi-task setting).

For your questions:

"Actually, I only reproduce the result of video-text: yc2c dataset." & "however, mine achieved the CIDEr score of 160.3! the original experiment CIDEr score is 153."

Do you mean the testing result reported from VALUE leaderboard under single-task setting?
Just like your observation "I only use the dataset of yc2c to train the model. so the trained model is more suitable to yc2c.", I think this is due to overfitting of the dataset.
And your analysis "I think it is the problem of dataset" also makes sense to me.

"Or is that possible that the result from poor experimental conditions can be better than results from advanced experimental conditions?"

Actually I don't think so. Using the same hyperparameters in different machines, the performance should be close.
Although we may have to reduce the batchsize due to machine limitations, we can achieve similar performance with gradient accumulation.

"I add the another dataset how2 to the yc2c, I found that is true that more dataset lowered the experiment results. how2+yc2c I got the CIDEr result of 151 to the yc2c. However, the model becomes more generalized."

We have not explored in depth for single-task learning for image-text or video-text experiments. As you can see from paper Table8, multi-task learning does help with generalizability and learning cross-task knowledge.
But how to prevent such knowledge forgetting during the multi-task learning is really an interesting topic!

Thanks for your issue!

**Comment by HenryHZY** [comment]
Thank you! I think we can close this issue:)

**Comment by Fiona-senior** [comment]
okSent from my iPhoneOn Oct 12, 2023, at 16:32, Zi-Yuan Hu ***@***.***> wrote:﻿
Thank you! I think we can close this issue:)
